plugin/semisync/semisync_master.h

```diff
@@ -439,7 +443,7 @@ class AckContainer : public Trace {
     if (m_ack_array) {
       for (unsigned i = 0; i < m_size; ++i) {
         m_ack_array[i].clear();
-        m_ack_array[i].server_id = 0;
+        m_ack_array[i].slave_uuid = "";
```

Comment:
I'm also seeing a crash here. I think it's because m_ack_array is allocated through my_malloc(), and the constructors are never called on it? Though if we keep server_id for this structure, then these changes and issues go away.